File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const map = new Map();

        for (let i = 0; i < nums.length; i++) {
            const potential = target - nums[i]
            if (map.has(potential)) return [map.get(potential), i]
            map.set(nums[i], i)
        }
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1

        while l < r:
            num1, num2 = numbers[l], numbers[r]
            currSum = num1 + num2

            if currSum == target:
                return [l + 1, r + 1]
            elif currSum > target:
                r -= 1
            else:
                l += 1
        
